### DISCUSSION TITLE
Debugging FastAPI apps with VSCode

### DISCUSSION THREAD

**Discussion by br3ndonland** [fastapi/fastapi#2556] [Show and tell]
@Kludex added a FastAPI debugger configuration to VSCode (microsoft/vscode-python#14247, microsoft/vscode-python#14606). Thank you @Kludex! :rocket: :clap:

Let's share our tips and troubleshooting here.

If you haven't tried it before, the [VSCode debugger](https://code.visualstudio.com/docs/python/debugging) is a great way to run your FastAPI app for local development. To use the debugger config:

- Update your vscode-python extension to version [2020.12.0](https://github.com/microsoft/vscode-python/blob/main/CHANGELOG.md#2020120-14-december-2020) or later, and reload VSCode.
- Navigate to the debugger pane, click "create a launch.json file," and select "FastAPI."

  <img width="1725" alt="vscode-python-debugger-fastapi-01-config" src="https://user-images.githubusercontent.com/26674818/102813266-85919c80-4396-11eb-9363-ac2b5b73ad55.png">

- Set breakpoints by clicking in the sidebar. Try setting a breakpoint in one of your FastAPI routes, then send a request to that route. VSCode will pause on your debugger breakpoint:

  <img width="2672" alt="vscode-python-debugger-fastapi-02-breakpoint" src="https://user-images.githubusercontent.com/26674818/102813394-bbcf1c00-4396-11eb-8eda-7d1cde53b3d6.png">

Feel free to clone the [project in the screenshot](https://github.com/br3ndonland/inboard) if you would like to try it out. You will need to install [Poetry](https://python-poetry.org/), then run `poetry install -E fastapi`. The theme in the screenshot is [Dracula Pro](https://draculatheme.com/pro), the font is [Dank Mono](https://gumroad.com/l/dank-mono), the shell is [Zsh with pure prompt](https://github.com/sindresorhus/pure), the HTTP client is [HTTPie](https://httpie.io/), and the VSCode settings are available [here](https://github.com/br3ndonland/dotfiles/blob/main/vscode/User/settings.json).

**Comment by RSA-2048** [comment]
Thanks, that was very helpful and worked like a charm!
